### PR TITLE
Blocked user profile name

### PR DESCRIPTION
**Pull request by TCbacon**

### CODE REVIEW

File: APILambda/user/getUser.test.ts [hunk 1]
```diff
@@ -37,3 +39,30 @@ describe("GetUser tests", () => {
...
   })
 })
+
+describe("Get blocked user's profile name only", () => {
+  resetDatabaseBeforeEach()
+
+  it("should return blocked user's profile name and themToYou status of blocked", async () => {
+    const fromUser = await global.registerUser({ name: "John Smith" })
+    const fromUserToken = await createUserAndUpdateAuth(fromUser)
+    const fromUserProfile = (await callGetSelf(fromUserToken)).body
```
Comment (seanim0920) [inline APILambda/user/getUser.test.ts]: can use fromUser.id and toUser.id

File: APILambda/user/getUser.test.ts [hunk 2]
```diff
@@ -37,3 +39,30 @@ describe("GetUser tests", () => {
     })
   })
 })
+
+describe("Get blocked user's profile name only", () => {
+  resetDatabaseBeforeEach()
+
+  it("should return blocked user's profile name and themToYou status of blocked", async () => {
+    const fromUser = await global.registerUser({ name: "John Smith" })
```
Comment (seanim0920) [inline APILambda/user/getUser.test.ts]: registerUser makes a new email in stage, use global.defaultUser

File: APILambda/user/getUser.ts [hunk 3]
```diff
@@ -51,25 +51,61 @@ const toUserWithRelationResponse = (user: DatabaseUserWithRelation) => ({
...
   )
 }
+
+const userAndRelationsWithId = (
+  conn: SQLExecutable,
+  userId: string,
+  fromUserId: string
+) =>
+  getProfileNameWithBlockStatus(conn, userId, fromUserId)
+    .flatMapSuccess((result) => {
```
Comment (seanim0920) [inline APILambda/user/getUser.ts]: can use mapSuccess().mapFailure() to remove manual promiseResult(...)

File: APILambda/user/getUser.ts [hunk 4]
```diff
@@ -51,25 +51,61 @@ const toUserWithRelationResponse = (user: DatabaseUserWithRelation) => ({
...
+        result.themToYouStatus === "blocked"
+      ) {
+        return promiseResult(success(result))
+      }
+      return promiseResult(failure("no result found" as const))
+    })
+    .flatMapFailure(() => {
+      return conn.queryFirstResult<DatabaseUserWithRelation>(
+        `
+      SELECT *,
```
Comment (seanim0920) [inline APILambda/user/getUser.ts:73]: there may be a way to do this without duplicating the query, but we can update that later

**Review by seanim0920 — DISMISSED** [review]

File: APILambda/user/getUser.ts [hunk 5]
```diff
@@ -51,25 +51,61 @@ const toUserWithRelationResponse = (user: DatabaseUserWithRelation) => ({
...
+  fromUserId: string
+) =>
+  getProfileNameWithBlockStatus(conn, userId, fromUserId)
+    .mapSuccess((result) => {
+      if (
+        result &&
+        Object.keys(result).length !== 0 &&
+        result.themToYouStatus === "blocked"
+      ) {
+        return result
```
Comment (TCbacon) [inline APILambda/user/getUser.ts]: Remove extra data if blocked is true

File: APILambda/user/getUser.ts [hunk 6]
```diff
@@ -51,25 +51,61 @@ const toUserWithRelationResponse = (user: DatabaseUserWithRelation) => ({
...
+  conn: SQLExecutable,
+  userId: string,
+  fromUserId: string
+) =>
+  getProfileNameWithBlockStatus(conn, userId, fromUserId)
+    .mapSuccess((result) => {
+      if (
+        result &&
+        Object.keys(result).length !== 0 &&
+        result.themToYouStatus === "blocked"
```
Comment (TCbacon) [inline APILambda/user/getUser.ts]: If blocked return profile picture, name, and handle

**Pull request merged**